client/src/lib/solana/sol-data.ts

```diff
@@ -323,58 +446,93 @@ export class DecentralizedIdentifier extends Assignable {
       : `${this.clusterType.toString()}:`;
     const urlField =
       !this.urlField || this.urlField === '' ? '' : `#${this.urlField}`; // TODO add support for / urls
-    return `${DID_HEADER}:${DID_METHOD}:${cluster}${this.pubkey.toString()}${urlField}`;
+    return `${DID_HEADER}:${DID_METHOD}:${cluster}${this.authorityPubkey.toString()}${urlField}`;
   }
 
   static REGEX = new RegExp('^did:' + DID_METHOD + ':?(\\w*):(\\w+)#?(\\w*)$');
 
-  static parse(did: string | DIDVerificationMethod): DecentralizedIdentifier {
-    if (typeof did === 'string') {
+  /**
+   * Parses a given did string
+   * @param did the did string
+   */
+  static async parse(
```

Comment:
This added async propagates quite far up through the code. Not too big of a deal, but I want to understand the introduced complexity here.

Reply:
Maybe I just don't expect a parser to do anything that needs to be awaited, but let's discuss on 1-on-1.

Reply:
I agree. Maybe we should rethink the DecetralizedIdentifier interface. Does it need the pdaPubKey in its constructor? it looks like it is inferred from authorityPubkey so it can be derived as needed instead